### DISCUSSION TITLE
v2.12.0

### DISCUSSION THREAD

**Discussion by jgraichen** [jgraichen/redmine_dashboard#213] [Announcements]
v2.12.0 contains several fixes and compatibility with Redmine 5.0 and Ruby 3.0 and 3.1:

### New
* Support for Ruby 3.0 and 3.1 with Redmine 5.0
* Support for Redmine 5.0
* Show or hide spend time and estimation based on the viewer's permissions (#90)

### Fixes
* Locale code file mapping for pt-BR and zh-TW
* Spacing and padding issues with compact card properties

##  Looking for testers!

Unfortunately, I do not have access to any larger Redmine project anymore, nor am I activity using Redmine anywhere anymore. I can only test with some artificially constructed local project that cannot represent any real project or usage. I would really appreciate if you can test v2.12.0 in your project or organization and report any findings (or their absence). Thank you!

<hr /><em>This discussion was created from the release <a href='https://github.com/jgraichen/redmine_dashboard/releases/tag/v2.12.0'>v2.12.0</a>.</em>

**Comment by dailson-igo** [comment]
Congratulations on the work done. 

I don't know exactly why, but I got the following error with Redmine 5, Ruby 3.1.2, Rails 6.1.5.1:

/home/deployruby/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.4/lib/zeitwerk/loader/callbacks.rb:25:in `on_file_autoloaded': expected file 
/opt/redmine5/plugins/redmine_dashboard/lib/rdb/rails/i18n.rb to define constant Rdb::Rails::I18n, but didn't (Zeitwerk::NameError)

      raise Zeitwerk::NameError.new("expected file #{file} to define constant #{cpath}, but didn't", cref.last)
      ^^^^^

Another good similar project, but that didn't work with Redmine 5 either, but works fine on Redmine 4, I found in the link:
https://github.com/nanego/redmine_scrum

Thanks.

**Reply by jgraichen** [reply]
I just release 2.12.1 with a fix. Can you please confirm that the error has been fixed?

Thanks for reporting and testing!

**Reply by dailson-igo** [reply]
Work fine with release 2.12.1.thank you very much, the correction was immediate.
Can I help you with release 3 remaining, Ruby 3.1 and Rails 6.1?

**Comment by dailson-igo** [comment]
Another question. When viewing the dashboard, "%[count]" appears instead of displaying the value with the amount of tasks with status of "new". Normalizes after I drag and drop some tasks.

**Comment by kyahagi** [comment]
After using it, an error may occur when browsing some projects in the following environment.

```
Environment:
  Redmine version                5.0.0.stable.21535
  Ruby version                   3.0.4-p208 (2022-04-12) [x86_64-linux]
  Rails version                  6.1.5
  Environment                    production
  Database adapter               PostgreSQL
  Mailer queue                   ActiveJob::QueueAdapters::AsyncAdapter
  Mailer delivery                smtp
Redmine settings:
  Redmine theme                  Classic
SCM:
  Subversion                     1.13.0
  Git                            2.25.1
  Filesystem                     
Redmine plugins:
  redmine_dashboard              2.12.1
  redmine_slack                  0.2
  view_customize                 3.0.0
```

■log■

```
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace] Processing by RdbTaskboardController#index as JS
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]   Parameters: {"_"=>"1652513841483", "id"=>"signz-for-biz"}
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]   Current user: XXXX (id=1)
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]   Rendered plugins/redmine_dashboard/app/views/rdb_dashboard/index.js.erb (Duration: 29.7ms | Allocations: 11396)
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace] Completed 500 Internal Server Error in 46ms (ActiveRecord: 10.2ms | Allocations: 15999)
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]   
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace] ActionView::Template::Error (translation data {:other=>"%{count} 件のチケット"} can not be used with :count => 1. key 'one' is missing.):
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]     4:       li.rdb-column class=(column.compact? ? 'rdb-column-compact' : '')
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]     5:         h3 = column.title
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]     6:         - if column.issues.count > 0
App 20861 output: [faa9d299-20b0-4049-acd9-2a5a87a58ace]     7:           span = t :rdb_x_issues, count: column.issues.coun
```
